### PR TITLE
Remove props to implement subrows

### PR DESCRIPTION
**Pull request by amanmahajan7**
- [x] Remove SubRow props
   - `getSubRowDetails`
   - `onCellExpand`
   - `onDeleteSubRow`
   - `onAddSubRow` - this prop was not used
- [x] Update `TreeView` example and implement subrows without the above props. SubRows implementation felt too complicated and I ended up using a simpler model
- [x] Combine `TreeView` and `TreeViewNoDelete` examples
- [x] Tsify `CellActions` example

Next step is to replace `defaultCellContentRenderer` with `defaultFormatter`

### CODE REVIEW

File: examples/demos/example27-cell-actions.tsx [hunk 1]
```diff
@@ -147,38 +146,19 @@ const columns = [
...
-        />
-      </Wrapper>
-    );
-  }
+  return (
+    <Wrapper title="Cell Actions Example">
+      <DataGrid<Row, 'id'>
+        enableCellSelect
+        columns={columns}
+        rowGetter={i => rows[i]}
```
Comment (nstepien) [inline examples/demos/example27-cell-actions.tsx]: It doesn't matter too much for demos, but we should avoid passing new functions on each render to DataGrid.

File: examples/demos/example25-tree-view.tsx [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+function toggleSubRow(rows: Row[], id: string): Row[] {
+  const rowIndex = rows.findIndex(r => r.id === id);
+  const row = rows[rowIndex];
+  const { children } = row;
+  if (!children) return rows;
+
+  const newRows = [...rows];
+  if (!row.isExpanded) {
+    newRows[rowIndex] = { ...newRows[rowIndex], isExpanded: true };
```
Comment (nstepien) [inline examples/demos/example25-tree-view.tsx]:
```suggestion
    newRows[rowIndex] = { ...row, isExpanded: true };
```
We could also move this line out of the if/else, and do
```suggestion
    newRows[rowIndex] = { ...row, isExpanded: !row.isExpanded };
```

File: examples/demos/example25-tree-view.tsx [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+    const subRows = children.map(c => ({ ...c }));
```
Comment (nstepien) [inline examples/demos/example25-tree-view.tsx]: Don't think we really need to clone the subrow objects.

File: examples/demos/example25-tree-view.tsx [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+    newRows.splice(rowIndex + 1, 0, ...subRows);
+  } else {
+    newRows[rowIndex] = { ...newRows[rowIndex], isExpanded: false };
```
Comment (nstepien) [inline examples/demos/example25-tree-view.tsx]: See above.

File: examples/demos/example25-tree-view.tsx (hunk 2 above)
Comment (amanmahajan7) [reply examples/demos/example25-tree-view.tsx]: Done

File: examples/demos/example27-cell-actions.tsx (hunk 1 above)
Comment (amanmahajan7) [reply examples/demos/example27-cell-actions.tsx]: I am thinking of using `eventBus` more aggressively and avoid passing callbacks down. I have a few ideas that we can discuss next week

**Pull request merged**